test/parallel/test-whatwg-readablestream.js

```diff
@@ -1290,3 +1312,211 @@ class Source {
     assert.strictEqual(stream[kState].state, 'closed');
   }));
 }
+
+{
+  assert.throws(() => Reflect.get(ReadableStream.prototype, 'locked', {}), {
+    code: 'ERR_INVALID_THIS',
+  });
```

Comment:
These kinds of tests should also be covered by the IDL harness tests... but I take it we haven't yet managed to make those run correctly in Node? 😞

Reply:
Once we get the IDL harness test working we can remove these. With your hack the IDL tests do work so maybe we can consider adding that to the wpt's?

Reply:
I don't think we should upstream this to WPT as-is. It's *really* hacky, and it may not be true for other environments. Maybe someone else wants to implement a `Window`-exposed interface in their project, and run these same IDL tests against their implementation.

However, I found a better way. We can add this to the `runner.setInitScript()` call:
```javascript
// For IDL harness tests, pretend that we're a DedicatedWorker.
global.DedicatedWorkerGlobalScope = global.constructor;
```
This makes us pass this check in [idlharness.js](https://github.com/web-platform-tests/wpt/blob/9e11b662c7204cf561801314bec46ec36220ebc6/resources/idlharness.js#L656-L659):
```javascript
if ('DedicatedWorkerGlobalScope' in self &&
    self instanceof DedicatedWorkerGlobalScope) {
    return globals.has("DedicatedWorker");
}
```
This is still a hack, but it's *much* less invasive and more flexible:
* We don't need to patch `idlharness.js`, so we can stay in sync with upstream WPT.
* We can put this in the specific test script itself (i.e. `test-streams.js`), so other tests aren't affected. In fact, other tests could use a similar hack if they want to test interfaces exposed on `Window` or `SharedWorker` or whatever. 😁

Thoughts?

Reply:
Hang on, that causes [`create_test_environment` in `testharness.js`](https://github.com/web-platform-tests/wpt/blob/master/resources/testharness.js#L475-L478) to create a `DedicatedWorkerTestEnvironment` instead of a `ShellTestEnvironment`. That environment uses `self.postMessage()` to send back the results back, so the tests never complete... Dangit. 😬

Reply:
Well, this works. But now it's leaning more towards the hacky side. 😛 
```javascript
Object.defineProperties(global, {
  DedicatedWorkerGlobalScope: {
    get() {
      // Pretend that we're a DedicatedWorker, but *only* for the IDL harness.
      // For the regular test harness, keep the JavaScript shell environment.
      if (new Error().stack.includes("idlharness.js")) {
        // self instanceof DedicatedWorkerGlobalScope === true
        return global.constructor;
      } else {
        // self instanceof DedicatedWorkerGlobalScope === false
        return function() {};
      }
    }
  }
});
```
Alternatively, we add a special case in [`test/common/wpt/worker.js`](https://github.com/nodejs/node/blob/ccc50a982044812e9e307a4626442c564cd60cf8/test/common/wpt/worker.js#L54) to temporarily add this global right before we load `idlharness.js`. But then it's shared across all WPT tests again... 😕 Or we have to add API to `WPTRunner` to choose which environment to emulate for the IDL harness.

Reply:
I think the defineProperties approach works fine for now. We can revisit this again later.

Reply:
Ok, IDL test is now working with the defineProperties. I'm going to go ahead and keep the additional checks I added in our own tests, just because they also check the error code.